gateway-service-impl/src/main/java/org/hypertrace/gateway/service/span/SpanService.java

```diff
@@ -185,46 +185,50 @@ private QueryRequest.Builder createQueryWithFilter(
   }
 
   private int getTotalFilteredSpans(RequestContext context, SpansRequest request) {
-    int total = 0;
-    String timestampAttributeId =
-        getTimestampAttributeId(
-            this.attributeMetadataProvider, context, AttributeScope.EVENT.name());
-
-    QueryRequest queryRequest =
-        createQueryWithFilter(request, context)
-            .addSelection(createCountByColumnSelection(timestampAttributeId))
-            .setLimit(1)
-            .build();
-
-    Iterator<ResultSetChunk> resultSetChunkIterator =
-        queryServiceClient.executeQuery(context, queryRequest);
-    while (resultSetChunkIterator.hasNext()) {
-      ResultSetChunk chunk = resultSetChunkIterator.next();
-      if (LOG.isDebugEnabled()) {
-        LOG.debug("Received chunk: " + chunk.toString());
-      }
-
-      // There should be only 1 result
-      if (chunk.getRowCount() != 1 && chunk.getResultSetMetadata().getColumnMetadataCount() != 1) {
-        LOG.error(
-            "Count the Api Traces total returned in multiple row / column. "
-                + "Total Row: {}, Total Column: {}",
-            chunk.getRowCount(),
-            chunk.getResultSetMetadata().getColumnMetadataCount());
-        break;
-      }
+    int total = -1;
+    if (request.getFetchTotal()) {
```

Comment:
We should add the check in the caller rather than here. We can either completely avoid setting total (which will make the total by default as 0), if the `request.getFetchTotal()` is false, or we can set it to -1

Reply:
Done.